rpcs3/rpcs3qt/user_manager_dialog.cpp

```diff
@@ -0,0 +1,359 @@
+
+	// Maybe we want a "Save and Quit" button, instead of saving immediately?
+	// Also, changing the user won't be written to memory until Emu.Init() is called,
+	// so you have to restart the emulator for it to take effect.
+	m_emu_settings->SetSetting(emu_settings::SelectedUser, selectedUserId);
+	m_emu_settings->SaveSettings(); 
+	UpdateTable();
+}
+
+void user_manager_dialog::OnSort(int logicalIndex)
+{
+	if (logicalIndex < 0)
+	{
+		return;
+	}
+	else if (logicalIndex == m_sort_column)
+	{
+		m_sort_ascending ^= true;
+	} 
+	else
+	{
+		m_sort_ascending = true;
+	}
+	Qt::SortOrder sort_order = m_sort_ascending ? Qt::AscendingOrder : Qt::DescendingOrder;
+	m_table->sortByColumn(m_sort_column, sort_order);
+	m_sort_column = logicalIndex;
+}
+
+void user_manager_dialog::ShowContextMenu(const QPoint &pos)
+{
+	QPoint globalPos = m_table->mapToGlobal(pos);
+	QMenu* menu = new QMenu();
+	int idx = m_table->currentRow();
+	if (idx < 0)
+	{
+		return;
+	}
+
+	// Create all the actions before adding them to the menu/submenus.
+	QAction* userIdAct = new QAction(tr("User ID"), this);
+	QAction* userNameAct = new QAction(tr("User Name"), this);
+
+	QAction* removeAct = new QAction(tr("&Remove"), this);
+	QAction* renameAct = new QAction(tr("&Rename"), this);
+	QAction* loginAct = new QAction(tr("&Login"), this);
+	QAction* showDirAct = new QAction(tr("&Open User Directory"), this);
+
+	//Create submenu for sort options.
+	m_sort_options = new QMenu(tr("&Sort By"));
+	m_sort_options->addAction(userIdAct);
+	m_sort_options->addAction(userNameAct);
+
+	// Add all options and submenus to the context menu.
+	menu->addMenu(m_sort_options);
+	menu->addSeparator();
+	menu->addAction(removeAct);
+	menu->addAction(renameAct);
+	menu->addAction(loginAct);
+	menu->addAction(showDirAct);
+
+	// Enable actions if selected user is not logged in user.
+	int idx_real = m_table->item(idx, 0)->data(Qt::UserRole).toInt();
+	bool enableButtons = m_user_list[idx_real]->GetUserId() != Emu.GetUsr();
+
+	removeAct->setEnabled(enableButtons);
+	renameAct->setEnabled(enableButtons);
+	loginAct->setEnabled(enableButtons);
+
+	// Connects and Events
+	connect(removeAct, &QAction::triggered, this, &user_manager_dialog::OnUserRemove);
+	connect(renameAct, &QAction::triggered, this, &user_manager_dialog::OnUserRename);
+	connect(loginAct, &QAction::triggered, this, &user_manager_dialog::OnUserLogin);
+	connect(showDirAct, &QAction::triggered, [=]()
+	{
+		int idx_real = m_table->item(idx, 0)->data(Qt::UserRole).toInt();
+		QString path = qstr(m_user_list[idx_real]->GetUserDir());
+		QDesktopServices::openUrl(QUrl("file:///" + path));
+	});
+
+	connect(userIdAct, &QAction::triggered, this, [=] {OnSort(0); });
+	connect(userNameAct, &QAction::triggered, this, [=] {OnSort(1); });
+
+	menu->exec(globalPos);
+}
+
+void user_manager_dialog::closeEvent(QCloseEvent * event)
+{
+	m_gui_settings->SetValue(gui::um_geometry, saveGeometry());
```

Comment:
add: `QDialog::closeEvent(event);` at the end and please remove the space between * and event in the argument